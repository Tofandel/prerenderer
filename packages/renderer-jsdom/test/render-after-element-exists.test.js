const path = require('path')
const Prerenderer = require('@prerenderer/prerenderer')
const Renderer = require('@prerenderer/renderer-jsdom')

const testDir = path.resolve(__dirname, '../../../tests/dynamic')

test('renders after #test-element is created, before should-already-be-rendered is dispatched', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    renderer: new Renderer({
      renderAfterElementExists: '#test-element',
    }),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/dynamic.html'])
  await prerenderer.destroy()
  expect(renderedRoutes[0].html).toContain('Render Output')
  expect(renderedRoutes[0].html).not.toContain('Oh no, it\'s too late!')
  expect(renderedRoutes).toMatchSnapshot()
})

test('renders after after #late-element exists', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    renderer: new Renderer({
      renderAfterElementExists: '#late-element',
    }),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/dynamic.html'])
  await prerenderer.destroy()
  expect(renderedRoutes[0].html).not.toContain('Render Output')
  expect(renderedRoutes[0].html).toContain('Oh no, it\'s too late!')
  expect(renderedRoutes).toMatchSnapshot()
})

test("time's-out if element doesn't exists after 2000ms", async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    renderer: new Renderer({
      timeout: 2000,
      renderAfterElementExists: '#non-existing',
    }),
  })

  await prerenderer.initialize()
  await expect(async () => {
    await prerenderer.renderRoutes(['/dynamic.html'])
  }).rejects.toThrowErrorMatchingSnapshot()
  await prerenderer.destroy()
})
