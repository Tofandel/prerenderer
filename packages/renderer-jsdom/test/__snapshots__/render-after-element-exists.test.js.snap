// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders after #test-element is created, before should-already-be-rendered is dispatched 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: Dynamic page</title>
</head>
<body><p id="test-element">Render Output</p></body></html>",
    "originalRoute": "/dynamic.html",
    "route": "/dynamic.html",
  },
]
`;

exports[`renders after after #late-element exists 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: Dynamic page</title>
</head>
<body><p id="late-element">Oh no, it's too late!</p></body></html>",
    "originalRoute": "/dynamic.html",
    "route": "/dynamic.html",
  },
]
`;

exports[`time's-out if element doesn't exists after 2000ms 1`] = `"Could not prerender: element '#non-existing' did not appear within 2s"`;
